### ISSUE TITLE
Is this project still supported?

### ISSUE THREAD

**Issue by misilot** [ExLibrisGroup/primo-explore-devenv#138]
Hello,

There haven't been any updates for almost 2 years now? I was wondering if this a supported development environment?

This is what I am getting when doing `npm install` 

```
23870 warn optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.2 (node_modules/browser-sync/node_modules/chokidar/node_modules/fsevents):
23871 warn notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm64"})
23872 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
23872 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
23872 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   linux
23872 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: arm64
23873 warn ws@8.2.3 requires a peer of bufferutil@^4.0.1 but none is installed. You must install peer dependencies yourself.
23874 warn ws@8.2.3 requires a peer of utf-8-validate@^5.0.2 but none is installed. You must install peer dependencies yourself.
23875 warn node-fetch@2.6.7 requires a peer of encoding@^0.1.0 but none is installed. You must install peer dependencies yourself.
23876 warn primo-explore-devenv@1.1.0 No license field.
23877 warn optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/fsevents):
23878 warn notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"arm64"})
23879 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
23879 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
23879 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   linux
23879 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: arm64
23880 verbose stack Error: canvas@2.9.1 install: `node-pre-gyp install --fallback-to-build`
23880 verbose stack Exit status 1
23880 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
23880 verbose stack     at EventEmitter.emit (events.js:189:13)
23880 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
23880 verbose stack     at ChildProcess.emit (events.js:189:13)
23880 verbose stack     at maybeClose (internal/child_process.js:970:16)
23880 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
23881 verbose pkgid canvas@2.9.1
23882 verbose cwd /home/node
23883 verbose Linux 5.10.104-linuxkit
23884 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
23885 verbose node v10.15.3
23886 verbose npm  v6.4.1
23887 error code ELIFECYCLE
23888 error errno 1
23889 error canvas@2.9.1 install: `node-pre-gyp install --fallback-to-build`
23889 error Exit status 1
23890 error Failed at the canvas@2.9.1 install script.
23890 error This is probably not a problem with npm. There is likely additional logging output above.
23891 verbose exit [ 1, true ]
```

**Comment by gloriousgeneralist** [comment]
Definitely still supported, though it shows its age often. Some people have more success with later versions of Node, for example mine worked with 14.17.0.

**Comment by Steve-A-Orr** [comment]
I agree that it shows it's age...very frequently. Unfortunately, they also don't seem to take Pull Requests from those interested in helping.

Ex Libris the least you could do to help those using your product is to keep up with the underlying technologies. For instance, you could move to  [**node v14** which is the _Maintenance LTS_ ](https://nodejs.org/en/about/releases/)and has EOL in 2023-04-30; or jump to the **v18** _Current_ and buy your time until 2025-04-30.

Or maybe the least you can do is to allow Pull Requests from trusted community members, so that they can do the work for you.

**Comment by misilot** [comment]
I don't know if it would make sense to possibly fork this project and more or less leave a note here (or as a new issue), kind of directing people to go to the new fork for an updated and maintained version.

**Comment by gloriousgeneralist** [comment]
I'm going to escalate this with Primo product management. I would also encourage more discussion on Slack since the developers follow there too at least sometimes.

**Comment by misilot** [comment]
@gloriousgeneralist Is there info any info on this slack space?

**Comment by gloriousgeneralist** [comment]
Here's an invite link, set to expire in 7 days https://join.slack.com/t/igelu-eluna-siwg/shared_invite/zt-1g2nga2vn-Of0i1ZVg77RYbgOJrOej8w

**Comment by misilot** [comment]
Thank you!

**Comment by kloevschall** [comment]
> I'm going to escalate this with Primo product management. I would also encourage more discussion on Slack since the developers follow there too at least sometimes.

I raised the issue during a meeting yesterday with product management and would recommend others to highlight the issue of the development environment not being kept up-to-date. It is an important part of the product support for Primo.

**Comment by ShoulM** [comment]
The project is working with Node v14. I am missing something?

**Comment by ajtag** [comment]
it might be worth looking at what is happening at leuven

https://github.com/mehmetc/devday22

**Comment by ShoulM** [comment]
I opened a new branch that will support Node v16.17.0 LTS([Node-v16.17.0-(LTS)-support](https://github.com/ExLibrisGroup/primo-explore-devenv/tree/Node-v16.17.0-(LTS)-support)). So if you want to use the latest node LTS pull from that branch. 
I tested it on PC and it seems to working ok (just needed to upgrade node-sass dependency). 
If I don't get any issues with this branch I will merge it with the master branch in the future.
